enhancements/machine-api/control-plane-machineset-vertical-scaling.md

```diff
@@ -0,0 +1,348 @@
+---
+title: control-plane-machineset-vertical-scaling
+authors:
+  - "@bergmannf"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers: 
+  - None
+creation-date: 2023-10-30
+last-updated: 2023-10-30
+tracking-link:
+  - https://issues.redhat.com/browse/OSD-15261
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Control plane machineset automatic vertical scaling
+
+## Summary
+
+To allow control plane nodes to remain at an adequate size for its cluster, this
+proposal introduces new configuration for the control plane machineset operator
+to allow it to make automated scaling decisions for control plane node sizing.
+
+## Motivation
+
+During normal operation of an OpenShift cluster, as worker node count and
+workloads increase, it becomes necessary to increase resources available to the
+control plane nodes.
+
+During SRE operations of the managed OpenShift product, there was already a
+trigger indentified, when these increases are required. Instead of requiring the
```

Comment:
What triggers are we talking about? Are the scale up/down triggers documented in this EP?

Reply:
The basis for the underlying OSD card are these SREP alerts: https://github.com/openshift/managed-cluster-config/blob/master/deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml 
However, based on the Slack discussion those would depend on Prometheus alerts & metrics, so this tries to recapture (and make configurable) the basis of the alerts.
Thinking about more triggers, it might be a good idea to also use something like https://docs.openshift.com/container-platform/4.13/scalability_and_performance/recommended-performance-scale-practices/recommended-control-plane-practices.html as a 'trigger' - especially worker node counts.